### PR TITLE
perf(hubble): improve libp2p sync

### PR DESCRIPTION
**Pull request by Wazzymandias**
## Motivation

- Update libp2p to latest version for bug fixes, features, and performance
  - Some specific examples include (but not limited to): [batch publishing](https://github.com/ChainSafe/js-libp2p-gossipsub/issues/344), tunable message concurrency, exported types for Message Cache to allow extension (e.g with Bloom Filter)
  - The latest release conforms to newest spec of libp2p, which has subtle but important improvements to gossipsub
 
## Change Summary

- Updated libp2p from 6.1.0 to 13.0.0
- Enabled batch publishing for significant reduction in Network I/O

## Merge Checklist

_Choose all relevant options below by adding an `x` now or at any time before submitting for review_

- [x] PR title adheres to the [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/) standard
- [x] PR has a [changeset](https://github.com/farcasterxyz/hub-monorepo/blob/main/CONTRIBUTING.md#35-adding-changesets)
- [x] PR has been tagged with a change label(s) (i.e. documentation, feature, bugfix, or chore)
- [ ] PR includes [documentation](https://github.com/farcasterxyz/hub-monorepo/blob/main/CONTRIBUTING.md#32-writing-docs) if necessary.
- [x] All [commits have been signed](https://github.com/farcasterxyz/hub-monorepo/blob/main/CONTRIBUTING.md#22-signing-commits)

### Before
image

### After
image

---

## PR-Codex overview
This PR updates dependencies, refactors PeerId imports, and improves IP family handling. 

### Detailed summary
- Updated Node.js version to 21 in CI workflow
- Refactored PeerId imports to use `@libp2p/interface`
- Improved IP family handling in `ipFamilyFromString` function
- Updated dependencies related to libp2p, libp2p-gossipsub, libp2p-noise, and more

> The following files were skipped due to too many changes: `apps/hubble/src/test/e2e/gossipNetwork.test.ts`, `apps/hubble/src/test/e2e/gossipNetworkBundle.test.ts`, `apps/hubble/src/hubble.ts`, `apps/hubble/src/network/p2p/gossipNode.test.ts`, `apps/hubble/src/network/p2p/gossipNode.ts`, `apps/hubble/src/network/p2p/gossipNodeWorker.ts`, `yarn.lock`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -58,9 +58,9 @@ jobs:
     strategy:
       matrix:
         include:
-          - node_version: 18
+          - node_version: 21
             runs_on: 'buildjet-8vcpu-ubuntu-2204'
-          - node_version: 20
+          - node_version: 21
```
Comment (adityapk00) [inline .github/workflows/ci.yml:63]: We should keep both 20 and 21 (but can remove 18) from the matrix

File: apps/hubble/src/network/p2p/protocol.ts [hunk 2]
```diff
@@ -17,7 +17,12 @@ export const msgIdFnStrictSign = (message: GossipSubMessage): Uint8Array => {
 };
 
 /* This has been imported from the libp2p-gossipsub implementation as it's not public there */
-export function msgIdFnStrictNoSign(msg: GossipSubMessage): Uint8Array {
+export async function msgIdFnStrictNoSign(msg: GossipSubMessage): Promise<Uint8Array> {
   // Hashes the raw message data
-  return noSignMsgId(msg.data);
+  const id = noSignMsgId(msg.data);
+  if (id instanceof Promise) {
```
Comment (adityapk00) [inline apps/hubble/src/network/p2p/protocol.ts:23]: we can just "await id" right, and javascript will handle both the promise and non-promise cases

File: apps/hubble/src/network/sync/syncEngine.ts [hunk 3]
```diff
@@ -1253,7 +1253,7 @@ class SyncEngine extends TypedEmitter<SyncEvents> {
     let fetchMessagesThreshold = HASHES_PER_FETCH;
     // If we have more messages but the hashes still mismatch, we need to find the exact message that's missing.
     if (ourNode && ourNode.numMessages >= 1) {
-      fetchMessagesThreshold = 1;
+      fetchMessagesThreshold = HASHES_PER_FETCH / 2;
```
Comment (adityapk00) [inline apps/hubble/src/network/sync/syncEngine.ts:1256]: Can you remove this line from this PR. We'll remove this whole concept from the DiffSync v2 PR

**Comment by adityapk00** [comment]
Pretty cool. I thought upgrading will be very messy, but looks quite managable.

**Comment by sds** [comment]
Closing due to age, and not actively being pursued. Thanks!

**Pull request closed without merging**